test/model/store_test.dart

```diff
@@ -130,18 +131,70 @@ void main() {
     await check(future).throws<AccountNotFoundException>();
   }));
 
+  test('GlobalStore.perAccount account is logged out while loading; then succeeds', () => awaitFakeAsync((async) async {
+    final globalStore = UpdateMachineTestGlobalStore(accounts: [eg.selfAccount]);
+    globalStore.prepareRegisterQueueResponse = (connection) =>
+      connection.prepare(
+        delay: TestGlobalStore.removeAccountDuration + Duration(seconds: 1),
+        json: eg.initialSnapshot().toJson());
+    final connection = globalStore.apiConnectionFromAccount(eg.selfAccount) as FakeApiConnection;
+    final future = globalStore.perAccount(eg.selfAccount.id);
+    check(connection.takeRequests()).length.equals(1); // register request
+
+    await logOutAccount(globalStore, eg.selfAccount.id);
+    check(globalStore.takeDoRemoveAccountCalls())
+      .single.equals(eg.selfAccount.id);
+
+    await check(future).throws<AccountNotFoundException>();
+    check(globalStore.takeDoRemoveAccountCalls()).isEmpty();
+    // no poll, server-emoji-data, or register-token requests
+    check(connection.takeRequests()).isEmpty();
+    check(connection).isOpen.isFalse();
+  }));
+
   test('GlobalStore.perAccount account is logged out while loading; then fails with HTTP status code 401', () => awaitFakeAsync((async) async {
-    final globalStore = LoadingTestGlobalStore(accounts: [eg.selfAccount]);
+    final globalStore = UpdateMachineTestGlobalStore(accounts: [eg.selfAccount]);
+    globalStore.prepareRegisterQueueResponse = (connection) =>
+      connection.prepare(
+        delay: TestGlobalStore.removeAccountDuration + Duration(seconds: 1),
```

Comment:
Because both `UpdateMachineTestGlobalStore` and `TestGlobalStore` share `removeAccountDuration` through `_DatabaseMixin`, perhaps we should find a new home to this constant. Maybe a "k"-prefixed const?

Reply:
Maybe; let's see if @gnprice has thoughts on that. I showed him in the office that a static `_DatabaseMixin.removeAccountDuration` wouldn't work (callers would have to access it using the mixin's name, like `FooMixin.removeAccountDuration`, and it doesn't make sense for the mixin to be public).

Reply:
Yeah, I think this is fine. If we find ourselves with more items like this, maybe we'll find some other way to organize them.